spec/middlewares/time_traveler_spec.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+describe TimeTraveler do
+  context "when called with a POST request" do
+    let(:app) { proc { [200, {}, %w[Live]] } }
+    let(:middleware) { described_class.new(app) }
+    let(:request) { Rack::MockRequest.new(middleware) }
+    let(:header_date) { "2021-01-02T03:04:05.000Z" }
+    let(:expected_date) { Time.zone.parse(header_date) }
+
+    context "when server date header passed" do
+      it "passes the request through unchanged" do
+        request.get("/", "X_WITH_SERVER_DATE" => header_date) do
+          expect(Time.zone.now).to eq(expected_date)
+        end
+      end
```

Comment:
Can we have another test here to show that date for other requests is unaffected?
```
      it "changes the current time on the server for a single request" do
        request.get("/", "X_WITH_SERVER_DATE" => header_date) do
          expect(Time.zone.now).to eq(expected_date)
        end

        request.get("/") do
          expect(Time.zone.now).to eq(some_other_date)
        end
      end
```
`some_other_date` being some set time that we set with `travel_to`?

Reply:
Apparently request block didn't work at all. Changed it to return time in the response and refactored contexts [here](66d81bb)